dspace-server-webapp/src/test/java/org/dspace/app/rest/AdvancedItemLevelVersioningIT.java

```diff
@@ -0,0 +1,384 @@
+        System.out.println("resolveToURLTest");
+
+        configurationService.setProperty("dspace.initialVersionSuffix", true);
+
+        context.turnOffAuthorisationSystem();
+
+        item = ItemBuilder.createItem(context, collection)
+                          .withTitle("Public item with first .1 suffix")
+                          .withIssueDate("2022-07-15")
+                          .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                          .withSubject("ExtraEntry")
+                          .build();
+
+        version1st = VersionBuilder.createVersion(
+            context, item, "First version of the item without .1 suffix")
+                .build();
+
+        System.out.println("hello from resolveToURLTest: " + item.getHandle());
+
+        // get URL with ".1" suffix
+        String resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        String resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+        String resolvedVersionHandle = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                          context, version1st.getItem().getHandle());
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, baseUrl + item.getHandle().split("\\.")[0]);
+        assertEquals(resolvedVersionHandle, baseUrl + version1st.getItem().getHandle());
+
+        // check that the resolving of the handle is still working
+        configurationService.setProperty("dspace.initialVersionSuffix", false);
+
+        // get URL with ".1" suffix
+        resolvedHandleWithSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                              context, item.getHandle());
+        // get URL without ".1" suffix
+        resolvedHandleWithoutSuffix = HandleServiceFactory.getInstance().getHandleService().resolveToURL(
+                                                context, item.getHandle().split("\\.")[0]);
+
+        System.out.println("resolved handle with suffix: " + resolvedHandleWithSuffix);
+        System.out.println("resolved handle without suffix: " + resolvedHandleWithoutSuffix);
+
+        assertEquals(resolvedHandleWithSuffix, baseUrl + item.getHandle());
+        assertEquals(resolvedHandleWithoutSuffix, resolvedHandleWithoutSuffix);
+    }
+
+    @Test
+    public void resolveToURLWithoutFirstSuffixTest() throws Exception {
```

Comment:
you should perform a test also with two versions to verify that the .1 version resolve correctly and the most recent handle resolve correctly too.